### ISSUE TITLE
attachPdfToInvoice bug

### ISSUE THREAD

**Issue by SerKnight** [XeroAPI/xero-node#338]
We have a problem when using the new OAuth 2.0 Node SDK. When we upload attachments to an invoice we can see in the notes that the attachment was processed correctly, but the invoice isn't actually there.

```
async function attachPdfToInvoice(xero, attachmentDetails, invoiceID) { 
  const pdf = fs.createReadStream('output.pdf') 
  return xero.accountingApi.createInvoiceAttachmentByFileName(xero.tenantIds[0], invoiceID, attachmentDetails.fileName, pdf)}
```

```
{ attachments: [ Attachment { attachmentID: 'f5055a50-ee0d-464e-92b9-0c7ae89ae205', fileName: 'DOE.pdf', url: 'https://api.xero.com/api.xro/2.0/Invoices/4feb697b-b1c7-44a8-8be2-6ee5d5cbdbb3/Attachments/DOE.pdf', mimeType: 'application/pdf', contentLength: 15018, includeOnline: undefined } ] }
```

You can see at the invoice https://go.xero.com/AccountsReceivable/View.aspx?InvoiceID=4feb697b-b1c7-44a8-8be2-6ee5d5cbdbb3 the pdf is not attached

**Comment by SerKnight** [comment]
Invoice attachments example shown here: https://github.com/XeroAPI/xero-node-oauth2-app/blob/master/src/app.ts#L968

Still need to verify multiple file types - will be finishing this one asap early next week!

**Comment by SerKnight** [comment]
Fixed in xero-node 4.1.7 release:

Sample code using upload: https://github.com/XeroAPI/xero-node-oauth2-app/blob/master/src/app.ts#L1017
